Sources/KeyboardSupport/UIScrollView+Inset.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  UIScrollView+Inset.swift
+//  KeyboardSupport-iOS
+//
+//  Created by Cuong Ngo on 7/24/18.
+//  Copyright © 2018 Bottle Rocket. All rights reserved.
+//
+
+import UIKit
+
+extension UIScrollView {
+    
+    private class Box: NSObject {
+        let value: UIEdgeInsets
+        
+        init(value: UIEdgeInsets) {
+            self.value = value
+        }
+    }
+    
+    private static var insetKey = "originalContentInset"
+    var originalContentInset: UIEdgeInsets? {
+        get {
+            guard let wrapper = objc_getAssociatedObject(self, &UIScrollView.insetKey) as? Box else { return nil }
```

Comment:
I've personally never really been a fan of using these associated objects. IMO, it feels like an Obj-C runtime hack that doesn't really have a place in a pure-Swift world.

Reply:
That's fair. Do you have a suggestion on what we could use instead for keeping track of the 'pre-keyboard' contentInset?

Reply:
I think we have two options if we want to avoid messing with the Obj-C runtime:
* Make it a requirement of the client to store the initial inset information via a property (similar to how a `KeyboardScrollable` needs the implementer to manually define a `keyboardWillShowObserver` and `keyboardWillHideObserver`).
* Create an object that will hold that information and require the client to use that object. This would be more of an overall architecture change that would take us down a path of providing a pre-baked object that could handle a lot of the keyboard management stuff rather than trying to get fancy with protocol extensions. I feel like this is the approach we'll eventually want to go since it's kind of annoying that we get some behavior for free via protocol extensions, but still are required to opt-in to some of the required properties (like the `keyboardWillShowObserver` mentioned above).

I'm fine keeping what we've got for now, with the expectation that @earlgaspard will fix it all in version `2.0.0`! 😉